logic/service/offline-service.js

```diff
@@ -871,31 +859,31 @@ exports.OfflineService = OfflineService = RawDataService.specialize(/** @lends O
      */
     deleteData: {
         value: function (objects, type, context) {
-            var self = this;
+            var self = this,
```

Comment:
There was a bug because you were using _this_ inside the closure.  I could have replaced the _this_ with _self_.  It seemed a six/half dozen kind of choice.